shell/platform/android/io/flutter/embedding/engine/FlutterOverlaySurface.java

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine;
+
+import android.view.Surface;
+
+public class FlutterOverlaySurface {
```

Comment:
This class needs the `@Keep` annotation. This is because the C++ code will do lookup by method/class name.  As a result, these names cannot change, but tools like Proguard or R8 make classes and methods names shorter to reduce code size.